proto/ibc/lightclients/solomachine/v1/solomachine.proto

```diff
@@ -67,3 +69,80 @@ message TimestampedSignature {
   uint64 timestamp                   = 2;
 }
 
+// SignBytes defines the signed bytes used for signature verification.
+message SignBytes {
+  option (gogoproto.goproto_getters) = false;
+
+  uint64 sequence    = 1;
+  uint64 timestamp   = 2;
+  string diversifier = 3;
+  // marshaled data
+  bytes data = 4;
```

Comment:
We have to be careful here that no two distinct messages (e.g. `PacketCommitmentData`, `PacketAcknowledgementData`) will ever serialise to the same bytes value. Does `proto` guarantee that (is the name of the message included in the serialised format somehow)?

Reply:
cc/ @aaronc @alexanderbez

Reply:
what about for `Any`s. Would two different proto types packed to an `Any` for the same interface guarantee distinct serialized bytes?

Reply:
@cwgoes, Proto does not do that. You can have two bytes, `Foo` and `Bar`, with the exact same fields s.t. if the values of those fields are the same, the byte representation is the same.

Reply:
Aye, let's track this with an issue then, https://github.com/cosmos/cosmos-sdk/issues/7277